pages/shoppingItems/[id].js

```diff
@@ -0,0 +1,85 @@
+import { useRouter } from "next/router";
+import { shoppingItems } from "../../lib/shoppingItemsData.js";
+import styled from "styled-components";
+import Image from "next/image";
+import Link from "next/link";
+
+export default function ShoppingItemDetails({ shoppingItem }) {
+  const router = useRouter();
+  const { id, backgroundColor, imageURL } = router.query;
```

Comment:
As already mentioned. If `backgroundColor `and `imageURL` are always available within the `shoppingItem` object (and this is the true because it is statically imported data), you can simplify your code by avoiding the use of query parameters and directly accessing the properties from `shoppingItem`. 

This approach would reduce complexity, improve readability, and make your component easier to maintain.

```suggestion
  const { id } = router.query;
```

Reply:
I have changed this.